crates/gitbutler-core/tests/virtual_branches/mod.rs

```diff
@@ -1266,28 +1224,21 @@ fn detect_mergeable_branch() -> Result<()> {
     virtual_branches::unapply_branch(project_repository, branch1_id)?;
     virtual_branches::unapply_branch(project_repository, branch2_id)?;
 
+    project_repository.repo().set_head("refs/heads/master")?;
     project_repository
-        .git_repository
-        .set_head(&"refs/heads/master".parse().unwrap())?;
-    project_repository
-        .git_repository
+        .repo()
         .checkout_head(Some(&mut git2::build::CheckoutBuilder::default().force()))?;
 
     // create an upstream remote conflicting commit
     std::fs::write(
         Path::new(&project.path).join(file_path),
         "line1\nline2\nline3\nline4\nupstream\n",
     )?;
-    commit_all(&project_repository.git_repository);
-    let up_target = project_repository
-        .git_repository
-        .head()
-        .unwrap()
-        .target()
-        .unwrap();
-    project_repository.git_repository.reference(
-        &"refs/remotes/origin/remote_branch".parse().unwrap(),
-        up_target.into(),
+    commit_all(project_repository.repo());
+    let up_target = project_repository.repo().head().unwrap().target().unwrap();
```

Comment:
We have to be careful with these `unwrap()` calls as a panic *may* cause a timeout in the UI. When we see them, let's `?` out of there or add `context()` in case of `Option`.

I think they might be from times when `thiserror` made it harder to error out at will.

Reply:
i only allowed myself to do it because its testcode

Reply:
🤦‍♂️
And I double checked that it's not test-code, yet I failed. Apologies.

There is a bunch of old unwraps still outside of test-code though, and I hope we can slowly eradicate them as touch the code for other refactors. Now it should be easy thanks to `anyhow`.